code/frontend/src/components/Answer/AnswerParser.tsx

```diff
@@ -7,23 +7,32 @@ type ParsedAnswer = {
     markdownFormatText: string;
 };
 
+let filteredCitations = [] as Citation[];
+
+// Define a function to check if a citation with the same Chunk_Id already exists in filteredCitations
+const isDuplicate = (citation: Citation) => {
+    return filteredCitations.some((c) => c.chunk_id === citation.chunk_id);
+};
+
 export function parseAnswer(answer: AskResponse): ParsedAnswer {
     let answerText = answer.answer;
     const citationLinks = answerText.match(/\[(doc\d\d?\d?)]/g);
 
     const lengthDocN = "[doc".length;
 
-    let filteredCitations = [] as Citation[];
+    filteredCitations = [] as Citation[];
     let citationReindex = 0;
     citationLinks?.forEach(link => {
         // Replacing the links/citations with number
         let citationIndex = link.slice(lengthDocN, link.length - 1);
         let citation = cloneDeep(answer.citations[Number(citationIndex) - 1]) as Citation;
-        if (!filteredCitations.find((c) => c.id === citationIndex)) {
+        if (!isDuplicate(citation) && !filteredCitations.find((c) => c.id === citationIndex)) {
```

Comment:
Can the `!filteredCitations.find((c) => c.id === citationIndex)` part be added to the `isDuplicate` function?

Reply:
Resolved